### PR TITLE
Handle and pass auth errors in callbacks

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/src/connector.rs [hunk 1]
```diff
@@ -141,18 +150,41 @@ impl Connector {
                                                     Some(base64_url::encode(&signed));
                                             }
                                             Err(e) => {
-                                                println!(
-                                                    "Nkey auth is disabled. sign error: {}",
-                                                    e
-                                                );
+                                                self.events_tx
```
Comment (caspervonb) [inline async-nats/src/connector.rs:153]: Altho, why doesn't this just bubble up as a regular error?

**Comment by Jarema** [comment]
@caspervonb refactored all callbacks into `event callback`. WDYT?

**Comment by Jarema** [comment]
@caspervonb refactored callbacks into events.

File: async-nats/src/options.rs [hunk 2]
```diff
@@ -429,154 +423,21 @@ impl ConnectOptions {
...
     /// ```
-    pub fn error_callback<F, Fut>(mut self, cb: F) -> ConnectOptions
+    pub fn event_callback<F, Fut>(mut self, cb: F) -> ConnectOptions
     where
-        F: Fn(ServerError) -> Fut + Send + Sync + 'static,
+        F: Fn(Event) -> Fut + Send + Sync + 'static,
         Fut: Future<Output = ()> + 'static + Send + Sync,
     {
-        self.error_callback =
-            CallbackArg1::<ServerError, ()>(Box::new(move |error| Box::pin(cb(error))));
```
Comment (caspervonb) [inline async-nats/src/options.rs:447]: Change the type to something like a concrete EventFn type?

Comment (Jarema) [reply async-nats/src/options.rs:447]: Was about removing it, but realized that this type is also used for JWT. Let's keep it.

**Review by caspervonb — APPROVED** [review]
lgtm

**Pull request merged**